### PR TITLE
docs: add securiy document

### PR DESCRIPTION
**Pull request by hbagdi**

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -47,7 +48,12 @@ and infrastructure.
 Please refer to [this document](concepts/ha-and-scaling.md) to understand
 failures scenarios, recovery methods, as well scaling considerations.
 
-## Guides and Tutorials
+### Security
+
+Please refere to [this document](concepts/security.md) to understand the
+default security settings and how to further secure the Ingress Controller.
+
+## Guides and tutorials
```
Comment (h3xar0n) [inline docs/README.md]: (Not sure if making it lowercase was deliberate, but it's fine if it was)

```suggestion
## Guides and Tutorials
```

File: docs/concepts/security.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+## Kubernetes RBAC
+
+Kong Ingress Controller is deployed with RBAC permissions as explained in
+[deployment](deployment.md) document.
+It has read and list permissions on most resources but requires update
+and create permission for a few resources to provide seamless integration.
+The permissions can be locked down further if needed depending on the specific
+use-case.
+This RBAC policy is associated with a ServiceAccount in Kubernetes, which
+takes of the Controller authenticating and authorizing against Kuberentes
```
Comment (h3xar0n) [inline docs/concepts/security.md]: Was this supposed to say "takes off"? I would rephrase it to avoid the gerunds. (Please clarify what it means first though.)

```suggestion
takes of the Controller authenticating and authorizing against Kuberentes
```

File: docs/concepts/security.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+API-server.
+
+## Kong Admin API protection
```
Comment (h3xar0n) [inline docs/concepts/security.md]: (If it's intentional, it's fine.)

```suggestion
## Kong Admin API Protection
```

File: docs/concepts/security.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+Kong's Admin API is used to control configuration of Kong and proxying behavior.
+If an attacker gets access to Kong's Admin API, Kong will be compromised
+and all bets are off at that point. Hence, it is important that the deployment
```
Comment (h3xar0n) [inline docs/concepts/security.md]: Instead of "all bets are off at that point," which does not sound sober/objective, describe what an attacker will be able to do. 

"An attacker will be able to perform the same actions as an authorized user--including actions that they may compromise the confidentiality, integrity, or availability of data proxied through Kong."

Something along those lines, more concise if possible.

File: docs/concepts/security.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+ensures that the likelyhood of this happening is minimized.
```
Comment (h3xar0n) [inline docs/concepts/security.md]:
```suggestion
ensures that the likelihood of this happening is minimized.
```

File: docs/concepts/security.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+In the example deployements, the Controller and Kong's Admin API communicate
+over the loopback(lo) interface of the pod. There is no authorization or
```
Comment (h3xar0n) [inline docs/concepts/security.md]:
```suggestion
over the loopback (`lo`) interface of the pod. There is no authorization or
```

File: docs/concepts/security.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+authentication being done over the loopback listner.
```
Comment (h3xar0n) [inline docs/concepts/security.md]:
```suggestion
authentication taking place over the loopback listener.
```

Comment (h3xar0n) [reply docs/concepts/security.md]: if possible to say _what_ is not doing authentication or authorization, it'd be better to make the sentence active. 

"_X_ is not performing authentication or authorization over the loopback listener."

File: docs/concepts/security.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+Although not ideal, this setup is simple to get started and can be further
```
Comment (h3xar0n) [inline docs/concepts/security.md]: Describe in objective (and ideally measurable) terms how something is simple.

```suggestion
Although not ideal, this setup requires fewer steps to get started and can be further
```

File: docs/concepts/security.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+hardened.
+
+Please note that it is very important that Kong's Admin API is not accessible
+inside the cluster as any malicious service can change Kong's configuration.
```
Comment (h3xar0n) [inline docs/concepts/security.md]: just to check--did you mean outside instead of inside?

File: docs/concepts/security.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+If you're exposing Kong's Admin API itself outside the cluster, please ensure
+that you have the necessary authentication in place first.
+
+### Authentication on Kong's Admin API
+
+If Kong's Admin API is protected with one of the authentication plugins,
+the Controller can authenticate itself against it to add another layer of
+security.
+The controller comes with support for injecting arbitrary HTTP headers
```
Comment (h3xar0n) [inline docs/concepts/security.md]:
```suggestion
The Controller comes with support for injecting arbitrary HTTP headers
```

File: docs/concepts/security.md [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+in the requests it makes to Kong's Admin API, which can be used to inject
+authentication credentials.
+The headers can be specified using the CLI flag `--admin-header` in the Ingress
+Controller.
+
+Ingress Controller will support mutual TLS based authentication on Kong's Admin
```
Comment (h3xar0n) [inline docs/concepts/security.md]: Just for consistency, see lines above.

(Compound modifier needs hyphens.) 

```suggestion
The Ingress Controller will support mutual-TLS-based authentication on Kong's Admin
```

File: docs/concepts/security.md [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+API in future.
+
+### Kong Enterprise RBAC
+
+Kong Enterprise comes with support for authentication and authorization on
+Kong's Admin API.
+
+Once a RBAC token is provisioned, Kong Ingress Controller can use the RBAC
```
Comment (h3xar0n) [inline docs/concepts/security.md]:
```suggestion
Once an RBAC token is provisioned, Kong Ingress Controller can use the RBAC
```

File: docs/concepts/security.md (hunk 9 above)
Comment (hbagdi) [reply docs/concepts/security.md]: Nope, inside is correct here.

What I mean to say:

Deploying Kong Admin API without any sort of authentication and open even inside a cluster or private network is a bad idea. You should always have authentication enabled on the Admin API or not expose it at all.

**Comment by hbagdi** [comment]
@h3xar0n update the PR based on the feedback. Please re-review. Thank you!

File: docs/concepts/security.md (hunk 9 above)
Comment (h3xar0n) [reply docs/concepts/security.md]: Understood, sounds good.

**Comment by hbagdi** [comment]
Thank you for the review @h3xar0n!
Much appreciated!
